File: services/content-service/src/main/java/com/novix/content_service/service/VideoService.java
package com.novix.content_service.service;

import com.novix.content_service.dto.request.CreateVideoRequest;
import com.novix.content_service.dto.request.UpdateVideoRequest;
import com.novix.content_service.dto.response.PageResponse;
import com.novix.content_service.dto.response.VideoResponse;
import com.novix.content_service.enums.VideoStatus;

public interface VideoService {
    VideoResponse createVideo(CreateVideoRequest request);

    VideoResponse getVideoByVideoId(String videoId);

    VideoResponse updateVideo(String videoId, UpdateVideoRequest request);

    void deleteVideo(String videoId);

    void updateVideoStatus(String videoId, VideoStatus status);

    PageResponse<VideoResponse> getAllPublicVideos(int page, int size);

    PageResponse<VideoResponse> getVideosByUploader(String uploaderUserId, int page, int size);

    PageResponse<VideoResponse> getVideosByCategory(Long categoryId, int page, int size);
}

File: frontend/admin/src/services/video.service.ts
import api from '@/lib/api';
import { Video, PageResponse, VideoStatus } from '@/types';

export const videoService = {
  async getVideos(page = 0, size = 20): Promise<PageResponse<Video>> {
    const response = await api.get<PageResponse<Video>>(`/api/v1/videos?page=${page}&size=${size}`);
    return response.data;
  },

  async getVideoById(videoId: string): Promise<Video> {
    const response = await api.get<Video>(`/api/v1/videos/${videoId}`);
    return response.data;
  },

  async updateVideoStatus(videoId: string, status: VideoStatus): Promise<void> {
    await api.patch(`/api/v1/videos/${videoId}/status?status=${status}`);
  },

  async deleteVideo(videoId: string): Promise<void> {
    await api.delete(`/api/v1/videos/${videoId}`);
  },
};
